renku_notebooks/api/classes/server.py

```diff
@@ -70,13 +71,13 @@ def __init__(
     def _check_flask_config(self):
         """Check the app config and ensure minimum required parameters are present."""
         if current_app.config.get("GITLAB_URL", None) is None:
-            raise ValueError(
-                "The gitlab URL is missing, it must be provided in "
+            raise ProgrammingError(
```

Comment:
Should this be its own error type, like `ConfigurationError` ?

Because the description of 
> The programming errors are bugs or unexpected cases. In the first case, they should lead
    to creating a new GitHub issue; in the latter, it may be necessary to handle the specific
    error to provide the user a precise explanation.

Doesn't really match this issue